crates/ruff_python_parser/src/lexer.rs

```diff
@@ -1318,25 +1321,74 @@ impl<'source> Lexer<'source> {
     fn token_start(&self) -> TextSize {
         self.token_range().start()
     }
+
+    pub(crate) fn take_value(&mut self) -> TokenValue {
+        std::mem::replace(&mut self.value, TokenValue::None)
+    }
+
+    /// Creates a checkpoint to which it can later return to using [Self::rewind].
+    pub(crate) fn checkpoint(&self) -> LexerCheckpoint<'src> {
+        LexerCheckpoint {
+            value: self.value.clone(),
+            current: self.current.clone(),
```

Comment:
```suggestion
            current: self.current,
```

I don't think the `clone` call is needed here because `Token` implements `Copy`